SYMBOL_TABLE_ENTRY_CREATION.md

```diff
@@ -0,0 +1,354 @@
+
+`GenSymEntry` is the basis of all server objects in Arkouda that represent one element. For example, the server side representation of a `pdarray`. When using `GenSymEntry` as the basis for these objects, all the properties are clearly defined by the single component that is being stored.
+
+These entries can also be used when an element consists of multiple components, but the components are necessary for understanding of a singular element. For example, `SegArraySymEntry` contains a `values` and a `segments` entry. Each of these are `pdarrays`, but `segments` is used to index into `values` and retrieve data. Here we have a very specific relation between the components which ensures that the properties of `GenSymEntry` make sense.
+
+Let's step through the `GenSymEntry` properties to get a sense of why objects like `SegArraySymEntry` inherit this class. 
+
+- `dtype` refers to the data type of the `values` entry. All data points in `values` are required to be of that type. `segments` is always required to be an `int` array and is not considered data.
+- `itemsize` again refers only to the size of a single data point in values. Additionally, because `segments` is always guaranteed to be of type `int` we are still able to perform size computations in relation to `segments`.
+- `size` refers to the number of data points in the `values` entry.
+- `ndim` will store the number of segments because this is equivalent to our number of dimensions.
+- `shape` currently stores the shape of the segments entry.
+
+As you can see, despite the entry containing multiple `SymEntry` objects its components still fit the format provided by `GenSymEntry`.
+
+<a id="parent-comp"></a>
+### When to use `CompositeSymEntry`
+Unlike `GenSymEntry`, `CompositeSymEntry` does not support objects containing only a single `SymEntry`. These objects are designed to handle objects containing multiple `SymEntry` objects that are not fully necessary to understand the object as whole. Additionally, these objects often have varying types of data between the `SymEntry` objects and unlike the `SegArraySymEntry` case, we do not know these types at compile time. Each `CompositeSymEntry` contains the properties below:
+
+- `ndim` stores the number of dimensions in the entry. Depending on the object format this may represent different things.
+- `size` stores the number of values in the entry. Again, this may represent different meaning depending upon the object.
+
+<a id="unsure"></a>
+### Still Unsure of which Parent Class to Use?
+If you are unsure of which parent class to utilize when creating a new Symbol Table entry class, contact the core development team for Arkouda. You can either post an issue in our [GitHub](https://github.com/Bears-R-Us/arkouda/issues) or send a message on [Gitter](https://gitter.im/ArkoudaProject/community#).
+
+<a id="build"></a>
+## Building the New Symbol Table Entry
```

Comment:
should we point out the file to edit is `MultiTypeSymEntry`?

Reply:
Yeah I added a line to initial blurb in the section to mention where to add.